p2p/src/simulated/network.rs

```diff
@@ -746,6 +751,10 @@ impl<P: PublicKey, F: SplitForwarder<P>> crate::Sender for SplitSender<P, F> {
         priority: bool,
     ) -> Result<Vec<P>, Error> {
         let recipients = (self.forwarder)(self.replica, &recipients, &message);
+        let Some(recipients) = recipients else {
+            // If the forwarder returns None, drop the message
+            return Ok(Vec::new());
```

Comment:
We don't hide that we didn't send to anyone (whoever is sending should already be able to tolerate no recipients being returned).